strongbox-db-schema/src/main/java/org/carlspring/strongbox/db/schema/StrongboxSchema.java

```diff
@@ -235,7 +238,182 @@ private void applySchemaChanges(JanusGraphManagement jgm)
+                          jgm.getVertexLabel(ARTIFACT),
+                          jgm.getVertexLabel(ARTIFACT_TAG));
+
+        jgm.addConnection(jgm.getEdgeLabel(ARTIFACT_GROUP_HAS_ARTIFACTS),
+                          jgm.getVertexLabel(ARTIFACT_ID_GROUP),
+                          jgm.getVertexLabel(ARTIFACT));
+
+        jgm.addConnection(jgm.getEdgeLabel(REMOTE_ARTIFACT_INHERIT_ARTIFACT),
+                          jgm.getVertexLabel(REMOTE_ARTIFACT),
+                          jgm.getVertexLabel(ARTIFACT));
+
+        jgm.addConnection(jgm.getEdgeLabel(ARTIFACT_COORDINATES_INHERIT_GENERIC_ARTIFACT_COORDINATES),
+                          jgm.getVertexLabel(RAW_ARTIFACT_COORDINATES),
+                          jgm.getVertexLabel(GENERIC_ARTIFACT_COORDINATES));
+
+        jgm.addConnection(jgm.getEdgeLabel(ARTIFACT_COORDINATES_INHERIT_GENERIC_ARTIFACT_COORDINATES),
+                          jgm.getVertexLabel(NUGET_ARTIFACT_COORDINATES),
+                          jgm.getVertexLabel(GENERIC_ARTIFACT_COORDINATES));
+
+        jgm.addConnection(jgm.getEdgeLabel(ARTIFACT_COORDINATES_INHERIT_GENERIC_ARTIFACT_COORDINATES),
+                          jgm.getVertexLabel(NPM_ARTIFACT_COORDINATES),
+                          jgm.getVertexLabel(GENERIC_ARTIFACT_COORDINATES));
+
+        jgm.addConnection(jgm.getEdgeLabel(ARTIFACT_COORDINATES_INHERIT_GENERIC_ARTIFACT_COORDINATES),
+                          jgm.getVertexLabel(MAVEN_ARTIFACT_COORDINATES),
+                          jgm.getVertexLabel(GENERIC_ARTIFACT_COORDINATES));
+
+        jgm.addConnection(jgm.getEdgeLabel(ARTIFACT_COORDINATES_INHERIT_GENERIC_ARTIFACT_COORDINATES),
+                          jgm.getVertexLabel(PYPI_ARTIFACT_COORDINATES),
+                          jgm.getVertexLabel(GENERIC_ARTIFACT_COORDINATES));
+
+    }
+
+    private void applyPropertyConstraints(JanusGraphManagement jgm)
+    {
+        // Vertex Property Constraints
+        addVertexPropertyConstraints(jgm,
+                                     ARTIFACT,
+                                     "uuid",
+                                     "storageId",
+                                     "repositoryId",
+                                     "created",
+                                     "lastUpdated",
+                                     "lastUsed",
+                                     "sizeInBytes",
+                                     "downloadCount",
+                                     "filenames",
+                                     "checksums");
+
+        addVertexPropertyConstraints(jgm,
+                                     REMOTE_ARTIFACT,
+                                     "uuid",
+                                     "cached",
+                                     "created");
+
+        addVertexPropertyConstraints(jgm,
+                                     GENERIC_ARTIFACT_COORDINATES,
+                                     "uuid",
```

Comment:
@anki2189 could you also please extract these property names into constants? Just like we have it with `Vertices` and `Edges`.

Reply:
Suggested changes are part of https://github.com/strongbox/strongbox-db/pull/20

Reply:
@anki2189 would you mind to move it here? It seems that here it is more suitable than within `ArtifactArchiveListing`.

Reply:
@sbespalov  Changes pushed, pls verify.